header-rewrite-filter/utility.cc

```diff
@@ -46,6 +46,27 @@ BooleanOperatorType StringToBooleanOperatorType(absl::string_view bool_operator)
     }
 }
 
+FunctionType StringToFunctionType(absl::string_view function) {
+    if (function.find(DYNAMIC_VALUE_HDR) == 0) {
+        return FunctionType::GetHdr;
+    } else if (function.find(DYNAMIC_VALUE_URL_PARAM) == 0) {
+        return FunctionType::Urlp;
+    } else {
+        return FunctionType::InvalidFunctionType;
+    }
+}
+
+absl::Status GetFunctionArgument(FunctionType function_type, std::string& function_expression){
+    try {
+        auto start = function_expression.find_first_of("(");
+        auto end = function_expression.find_last_of(")");
+        function_expression = function_expression.substr(start+1, function_expression.size()-start-2);
+        return absl::OkStatus();
+    } catch (std::exception& e) {
+        return absl::InvalidArgumentError("failed to get function argument");
+    }
+}
```

Comment:
This is a perfect candidate for unit tests: it's a pure function, and it's doing the kind of string indexing that's very prone to off-by-one errors

Reply:
agreed, we should add tests to this PR